csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.shared.cs

```diff
@@ -422,8 +425,70 @@ public void RegisterCustomOpLibrary(string libraryPath)
         /// <param name="libraryHandle">out parameter, library handle</param>
         public void RegisterCustomOpLibraryV2(string libraryPath, out IntPtr libraryHandle)
         {
+            // NOTE: This is confusing due to the history.
+            // SessionOptions.RegisterCustomOpLibrary initially called NativeMethods.OrtRegisterCustomOpsLibrary
+            // and leaked the handle.
+            // SessionOptions.RegisterCustomOpLibraryV2 was added to resolve that by returning the handle.
+            // Later, NativeMethods.OrtRegisterCustomOpsLibrary_V2 was added with ORT owning the handle.
+            //
+            // End result of that is
+            //   SessionOptions.RegisterCustomOpLibrary calls NativeMethods.OrtRegisterCustomOpsLibrary_V2
+            //   SessionOptions.RegisterCustomOpLibraryV2 calls NativeMethods.OrtRegisterCustomOpsLibrary
             var utf8Path = NativeOnnxValueHelper.StringToZeroTerminatedUtf8(libraryPath);
-            NativeApiStatus.VerifySuccess(NativeMethods.OrtRegisterCustomOpsLibrary(handle, utf8Path, out libraryHandle));
+            NativeApiStatus.VerifySuccess(NativeMethods.OrtRegisterCustomOpsLibrary(handle, utf8Path, 
+                                                                                    out libraryHandle));
+        }
+
+        /// <summary>
+        /// Register custom ops by calling the C function with functionName. 
+        /// The C function has the signature
+        ///   OrtStatus* RegisterCustomOps(OrtSessionOptions* options, const OrtApiBase* api);
+        /// 
+        /// The function must be available in the global symbols so ONNX Runtime can find it using GetProcAddress or
+        /// dlsym. This will require your app to 'DllImport' the native library containing the function and use 
+        /// Marshal.Prelink to ensure the function is loaded.
+        ///
+        /// Example usage:
+        ///   static class NativeCustomOps {
+        ///     [DllImport("YourLibraryName", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Winapi)]
+        ///     public static extern IntPtr /* OrtStatus* */ 
+        ///         YourRegisterCustomOpsFunctionName(IntPtr /* OrtSessionOptions* */ sessionOptions,
+        ///                                           IntPtr /* OrtApiBase*        */ ortApiBase);
+        ///   }
+        ///   ...
+        ///   var sessionOptions = new SessionOptions();
+        ///   Marshal.Prelink(typeof(NativeCustomOps).GetMethod("YourRegisterCustomOpsFunctionName"));
```

Comment:
Can we add a helper function of implementing the static class and Marshal.Prelink to make it be easier to use?

Reply:
I don't think so. The library name needs to be used in the DllImport attribute.